### ISSUE TITLE
`nvim_oxi::api::open_win()` returning config validation error

### ISSUE THREAD

**Issue by dasupradyumna** [noib3/nvim-oxi#202]
## Description

The below snippet of code is returning a `Validation` error involving the **bufpos** field when trying to open either a float or a horizontal split. The debug output shows that **bufpos** field is `None` in both cases, yet the error is raised.  
This behavior can be reproduced for a vertical split as well.

```rust
// case 1 : hsplit
let config = WindowConfig.builder().split(SplitDirection::Below).height(10).build();
// case 2 : float
let config = WindowConfig.builder()
    .relative(api::types::WindowRelativeTo::Editor)
    .row(10)
    .col(10)
    .width(80)
    .height(30)
    .style(api::types::WindowStyle::Minimal)
    .border(api::types::WindowBorder::Rounded)
    .build();   
nvim_oxi::dbg!(&config);
let window = api::open_win(&buffer, true, &config)?; // this line returns the error out of this function
```

Is there any error in my usage of the API or is it a bug?

## Screenshots

The errors in question -

1. Horizontal split case
![image](https://github.com/user-attachments/assets/d131e7ce-a2cd-4319-a23e-67b9e77bdf95)

2. Floating window case
![image](https://github.com/user-attachments/assets/a43f477c-bb8f-4886-a2c8-3c3190492a01)

## Versions

**nvim-oxi** = { version = `0.5`, features = `neovim-0-10` }

**Comment by noib3** [comment]
Can you reproduce it with `main`?

**Comment by dasupradyumna** [comment]
Yes, I was able to reproduce this behavior on `main` with the same features `neovim-0-10` attribute.

**Comment by noib3** [comment]
Are you sure you have Neovim 0.10 in your `$PATH`? If so, can you write a test that fails on 0.10? I just tried reproducing this [here](https://github.com/noib3/nvim-oxi/blob/main/tests/src/api/win_config.rs#L3-L12), but the test passes.

**Comment by dasupradyumna** [comment]
It seems that my Neovim was updated to 0.11 by **apt** manager.
Changing features to `neovim-nightly` fixed the issue!

Thank you for all the help and my apologies for wasting your time with this non-issue.

**Comment by dasupradyumna** [comment]
I am facing the same issue with `nvim-oxi = { version = "0.5.1", features = ["neovim-nightly"] }`, with my neovim also being nightly. The issue does not occur when using **main** branch. Any idea why?

**Comment by noib3** [comment]
Because `main` contains fixes that `0.5.1` doesn't.

**Comment by dasupradyumna** [comment]
Okay, I'll stick to `main` until `v0.5.2` is released. Thank you!
